tests/unit/iontests.js

```diff
@@ -11,126 +11,365 @@
+            'good/intLongMaxValuePlusOne.10n',
+            'good/intLongMinValue.10n',
+            'good/nopPadInsideStructWithNopPadThenValueNonZeroSymbolId.10n',
+            'good/nopPadInsideStructWithNopPadThenValueZeroSymbolId.10n',
+            'good/nopPadInsideStructWithValueThenNopPad.10n',
+            'good/structAnnotatedOrdered.10n',
+            'good/structOrdered.10n',
+            'good/structUnordered.10n',
+            'good/timestamp/timestamp2011-02-20.10n',
+            'good/timestamp/timestamp2011-02-20T19_30_59_100-08_00.10n',
+            'good/timestamp/timestamp2011-02.10n',
+            'good/timestamp/timestamp2011.10n',
+            'good/equivs/systemSymbols.ion',//IVM
+            'bad/boolWithInvalidLength_2.10n',
+            'bad/boolWithInvalidLength_1.10n',
+            'bad/emptyAnnotatedInt.10n',
+            'bad/listWithValueLargerThanSize.10n',
+            'bad/localSymbolTableWithMultipleSymbolsFields.10n',
+            'bad/minLongWithLenTooLarge.10n',
+            'bad/minLongWithLenTooSmall.10n',
+            'bad/negativeIntZero.10n',
+            'bad/negativeIntZeroLn.10n',
+            'bad/nopPadTooShort.10n',
+            'bad/nopPadWithAnnotations.10n',
+            'bad/stringLenTooLarge.10n',
+            'bad/stringWithLatinEncoding.10n',
+            'bad/structOrderedEmpty.10n',
+            'bad/annotationLengthTooShortScalar.10n',
+            'bad/annotationLengthTooShortContainer.10n',
+            'bad/annotationNested.10n',
+            'bad/timestamp/timestampHourWithoutMinute.10n',
+
+
+
+        ];
+
+        // For debugging, put single files in this list to have the test run only
+        // that file.   (But don't forget to clean this up on checkin!)
+        let debugList = [
+            //'good/sexps.ion'
+        ];
+
+        let goodUnskipped = [];
+        for (let path of goodAccumulator) {
+            let spath = path.replace(/\\/g, "/");
+            let shouldSkip = false;
+            for (let skip of skipList) {
+                if (spath.endsWith(skip)) {
+                    shouldSkip = true;
+                    break;
+                }
+            }
+            if (debugList.length > 0) {
+                shouldSkip = true;
+                for (let debug of debugList) {
+                    if (spath.endsWith(debug)) {
+                        shouldSkip = false;
+                        break;
+                    }
+                }
+            }
+            if (!shouldSkip) {
+                goodUnskipped.push(path);
+            }
         }
-      }
-      if (!shouldSkip) {
-        unskipped.push(path);
-      }
-    }
 
-    var suite = {
-      name: 'Good tests',
-    };
-
-    function exhaust(reader) {
-      var tries = 0;
-      for (;;) {
-        // Safety valve
-        tries++;
-        if (tries > 1000) {
-          throw new Error("Reader spinning forever!");
+        let badUnskipped = [];
```

Comment:
Looks like a good opportunity to DRY this up. Trivially different than the `good` case above.

Reply:
I'm not familiar with that term

Reply:
Don't Repeat Yourself